File: frameworks/test.persistence/src/main/java/com/example/scaffold/frameworks/test/persistence/Persistences.java
package com.example.scaffold.frameworks.test.persistence;

import java.util.Map;
import java.util.function.Supplier;

@SuppressWarnings("all")
public final class Persistences {

    private Persistences() {
    }

    private static final Map<Class<?>, Supplier<Object>> PERSISTENCE_MAP = Map.of(
            TestClient.class, testClient()
    );

    public static <T> T fetchByType(Class<T> clazz) {
        return (T) PERSISTENCE_MAP.get(clazz).get();
    }

    public static boolean exist(Class<?> clazz) {
        return PERSISTENCE_MAP.containsKey(clazz);
    }

    private static Supplier<Object> testClient() {
        return () -> new TestClient();
    }

}

File: frameworks/test.persistence/src/main/kotlin/com/qomolangma/frameworks/test/persistence/Persistences.kt
@file:Suppress("UNCHECKED_CAST")

package com.qomolangma.frameworks.test.persistence

import java.util.function.Supplier

object Persistences {
    private val PERSISTENCE_MAP = mapOf<Class<*>, Supplier<Any>>(
        Pair(TestClient::class.java, testClient())
    )

    fun <T> fetchByType(clazz: Class<T>): T {
        return PERSISTENCE_MAP[clazz]!!.get() as T
    }

    fun exist(clazz: Class<*>): Boolean {
        return PERSISTENCE_MAP.containsKey(clazz)
    }

    private fun testClient(): Supplier<Any> {
        return Supplier { TestClient() }
    }
}
